dev/tools/localization/bin/gen_localizations.dart

```diff
@@ -562,17 +579,35 @@ void main(List<String> rawArgs) {
   );
 
   try {
+    validateLocalizations(widgetsLocaleToResources, widgetsLocaleToResourceAttributes, removeUndefined: options.removeUndefined);
     validateLocalizations(materialLocaleToResources, materialLocaleToResourceAttributes, removeUndefined: options.removeUndefined);
     validateLocalizations(cupertinoLocaleToResources, cupertinoLocaleToResourceAttributes, removeUndefined: options.removeUndefined);
   } on ValidationError catch (exception) {
     exitWithError('$exception');
   }
-
   if (options.removeUndefined) {
+    removeUndefinedLocalizations(widgetsLocaleToResources);
     removeUndefinedLocalizations(materialLocaleToResources);
     removeUndefinedLocalizations(cupertinoLocaleToResources);
   }
 
+  final String? widgetsLocalizations = options.writeToFile || !options.cupertinoOnly
+      ? generateArbBasedLocalizationSubclasses(
+        localeToResources: widgetsLocaleToResources,
+        localeToResourceAttributes: widgetsLocaleToResourceAttributes,
+        generatedClassPrefix: 'WidgetsLocalization',
+        baseClass: 'GlobalWidgetsLocalizations',
+        generateHeader: generateWidgetsHeader,
+        generateConstructor: generateWidgetsConstructor,
+        generateConstructorForCountrySubClass: generateWidgetsConstructorForCountrySubclass,
+        factoryName: widgetsFactoryName,
+        factoryDeclaration: widgetsFactoryDeclaration,
+        callsFactoryWithConst: true,
+        factoryArguments: widgetsFactoryArguments,
+        supportedLanguagesConstant: widgetsSupportedLanguagesConstant,
+        supportedLanguagesDocMacro: widgetsSupportedLanguagesDocMacro,
+      )
+      : null;
```

Comment:
What are the exact differences between how widgets localizations are generated and the other material/cupertino localizations? I'm seeing that we're using a different constructor `generateWidgetsConstructorForCountrySubclass` but I'm not sure why we need to do this.

Reply:
previously generateArbBasedLocalizationSubclasses forced the generated localization class constructor to have the same function signature with the base class.

It would generate something like this

class WidgetLocalizationES extends GlobalWidgetLocalization {
  class WidgetLocalizationEn(super.somparameters);
}

class WidgetLocalizationESMX extends WidgetLocalizationES {
  class WidgetLocalizationESMX(super.somparameters);
}

This makes it a bit inconvenient. that the WidgetLocalizationES and WidgetLocalizationESMX must always take in the same parameters. generateWidgetsConstructorForCountrySubclass is used when generating the constructor of WidgetLocalizationESMX so that it can have different function signature than WidgetLocalizationES.

Of curse I can make it all take the same parameter(in this case it would be TextDirection). but have it hard code directly in the WidgetLocalizationES would be more performant.